SoftLayer/managers/block.py

```diff
@@ -0,0 +1,317 @@
+"""
+    SoftLayer.block
+    ~~~~~~~~~~~~~~~
+    Block Storage Manager
+
+    :license: MIT, see LICENSE for more details.
+"""
+from SoftLayer import exceptions
+from SoftLayer import utils
+
+
+class BlockStorageManager(utils.IdentifierMixin, object):
+    """Manages Block Storage volumes."""
+
+    def __init__(self, client):
+        self.configuration = {}
+        self.client = client
+        self.account = client['Account']
+        self.product_package = self.client['Product_Package']
+        self.block_svc = self.client['Network_Storage']
+        self.block_os_types = self.client['Network_Storage_Iscsi_OS_Type']
+        self.product_order = self.client['Product_Order']
+
+    def list_block_volumes(self, datacenter=None, username=None,
+                           storage_type=None, **kwargs):
+        """Returns a list of block volumes.
+
+        :param datacenter: Datacenter short name (e.g.: dal09)
+        :param username: Name of volume.
+        :param storage_type: Type of volume: Endurance or Performance
+        :param kwargs:
+        :return: Returns a list of block volumes.
+        """
+        if 'mask' not in kwargs:
+            items = [
+                'id',
+                'username',
+                'capacityGb',
+                'bytesUsed',
+                'serviceResource.datacenter[name]',
+                'serviceResourceBackendIpAddress'
+            ]
+            kwargs['mask'] = "mask[%s]" % ','.join(items)
+
+        call = 'getIscsiNetworkStorage'
+
+        _filter = utils.NestedDict(kwargs.get('filter') or {})
+
+        _filter['iscsiNetworkStorage']['serviceResource']['type']['type'] = \
+            (utils.query_filter('!~ ISCSI'))
+
+        _filter['iscsiNetworkStorage']['storageType']['keyName'] = (
+            utils.query_filter('*BLOCK_STORAGE'))
+        if storage_type:
+            _filter['iscsiNetworkStorage']['storageType']['keyName'] = (
+                utils.query_filter('%s_BLOCK_STORAGE' % storage_type.upper()))
+
+        if datacenter:
+            _filter['iscsiNetworkStorage']['serviceResource']['datacenter'][
+                'name'] = (utils.query_filter(datacenter))
+
+        if username:
+            _filter['iscsiNetworkStorage']['username'] = \
+                (utils.query_filter(username))
+
+        kwargs['filter'] = _filter.to_dict()
+        func = getattr(self.account, call)
+        return func(**kwargs)
```

Comment:
To reduce the noise, this can be simplified to:

```python
return self.client.call('Account', 'getIscsiNetworkStorage', **kwargs)
```

Which removes the need to use getattr, and removes the "call" variable.